### ISSUE TITLE
Bayes grid estimation is not called when sometimes necessary

### ISSUE THREAD

**Issue by paulbarber** [flimlib/flimj-ops#11]
Bayes results are based on searching a multidimensional grid of possible values and then refining from there. This grid needs to be setup to cover the possible range of parameters (lifetime etc.)

In BayesFitWorker.java
On creation of BayesFitWorker class instance BayesFitWorker() looks for min/max == NULL then uses estimategrid() and calls Bayes_set _search_grid() in FLIMlib to set the grid.
So on creation the grid is estimated and set and seems to work with the test data.
Is this done at any other time, or is there a way to force it when the estimation is not correct?
Not sure of the optimal solution. In TRI2 we have an "Estimate Cursors" button which will also estimates the grid. This is where I would look at the sum of entire image transient and estimate the grid on that.

**Comment by Xanthorapedia** [comment]
> Is this done at any other time, or is there a way to force it when the estimation is not correct?

No. This is performed only once at the start by each worker. But I don't see a problem with redoing the estimation (and probably start over) in the middle of a dataset fitting.

> In TRI2 we have an "Estimate Cursors" button which will also estimates the grid. This is where I would look at the sum of entire image transient and estimate the grid on that.

That would be possible. The current way is to do an RLD and look at fits with 20% lowest chi^2 to estimate the mean and std to set the ranges.

How would you suggest we improve? Thanks.
